components/kyma-environment-broker/internal/event-hub/event_hub_step.go

```diff
@@ -0,0 +1,209 @@
+	return &ProvisionAzureEventHubStep{
+		operationManager:    process.NewOperationManager(os),
+		accountProvider:     accountProvider,
+		context:             ctx,
+		hyperscalerProvider: hyperscalerProvider,
+	}
+}
+
+func (p *ProvisionAzureEventHubStep) Name() string {
+	return "Provision Azure Event Hubs"
+}
+
+func (p *ProvisionAzureEventHubStep) Run(operation internal.ProvisioningOperation,
+	log logrus.FieldLogger) (internal.ProvisioningOperation, time.Duration, error) {
+
+	hypType := hyperscaler.Azure
+
+	// parse provisioning parameters
+	pp, err := operation.GetProvisioningParameters()
+	if err != nil {
+		// if the parameters are incorrect, there is no reason to retry the operation
+		// a new request has to be issued by the user
+		log.Errorf("Aborting after failing to get valid operation provisioning parameters: %v", err)
+		return p.operationManager.OperationFailed(operation, "invalid operation provisioning parameters")
+	}
+	log.Infof("HAP lookup for credentials to provision cluster for global account ID %s on Hyperscaler %s", pp.ErsContext.GlobalAccountID, hypType)
+
+	// get hyperscaler credentials from HAP
+	credentials, err := p.accountProvider.GardenerCredentials(hypType, pp.ErsContext.GlobalAccountID)
+	if err != nil {
+		errorMessage := fmt.Sprintf("Unable to retrieve Gardener Credentials from HAP lookup: %v", err)
+		// if gardener pool is not configured, retries won't solve the problem
+		return p.operationManager.OperationFailed(operation, errorMessage)
+	}
+	azureCfg, err := azure.GetConfigfromHAPCredentialsAndProvisioningParams(credentials, pp)
+	if err != nil {
+		// internal error, repeating doesn't solve the problem
+		errorMessage := fmt.Sprintf("Failed to create Azure config: %v", err)
+		return p.operationManager.OperationFailed(operation, errorMessage)
+	}
+
+	// create hyperscaler client
+	namespaceClient, err := p.hyperscalerProvider.GetClient(azureCfg)
+	if err != nil {
+		// internal error, repeating doesn't solve the problem
+		errorMessage := fmt.Sprintf("Failed to create Azure EventHubs client: %v", err)
+		return p.operationManager.OperationFailed(operation, errorMessage)
+	}
+
+	// create Resource Group
+	groupName := pp.Parameters.Name
+	// TODO(nachtmaar): use different resource group name https://github.com/kyma-incubator/compass/issues/967
+	resourceGroup, err := namespaceClient.CreateResourceGroup(p.context, azureCfg, groupName)
+	if err != nil {
+		// retrying might solve the issue while communicating with azure, e.g. network problems etc
+		errorMessage := fmt.Sprintf("Failed to persist Azure Resource Group [%s] with error: %v", groupName, err)
+		return p.retryOperation(operation, errorMessage, time.Minute, time.Minute*30, log)
+	}
+	log.Printf("Persisted Azure Resource Group [%s]", groupName)
+
+	// create EventHubs Namespace
+	eventHubsNamespace := pp.Parameters.Name
+	eventHubNamespace, err := namespaceClient.CreateNamespace(p.context, azureCfg, groupName, eventHubsNamespace)
```

Comment:
to be sure, if the method will be called second time with the same params, update operation will be run and no error will be return?

Reply:
Good catch!

Reply:
if the method is called for the second time with the same parameter, it will not return any error. it would just try to update it.  
ref: https://github.com/Azure/azure-sdk-for-go/blob/f1b3b9b7d4962c94c77d16ce75f9a8e981eb2488/services/eventhub/mgmt/2017-04-01/eventhub/namespaces.go#L132